torchvision/datasets/svhn.py

```diff
@@ -66,7 +66,16 @@ def __init__(self, root, split='train',
         loaded_mat = sio.loadmat(os.path.join(root, self.filename))
 
         self.data = loaded_mat['X']
-        self.labels = loaded_mat['y']
+        # loading from the .mat file gives an np array of type np.uint8
+        # converting to np.int64, so that we have a LongTensor after
+        # the conversion from the numpy array
+        # the squeeze is needed to obtain a 1D tensor
+        self.labels = loaded_mat['y'].astype(np.int64).squeeze()
+
+        # the svhn dataset assigns the class label "10" to the digit 0
+        # this makes it inconsistent with several loss functions
+        # which expect the class labels to be in the range [0, C-1]
+        np.place(self.labels, self.labels == 10, 0)
```

Comment:
Not sure if this is a good idea for 2 reasons:
1) this change will break backwards compatibility
2) this will make the labels returned inconsistent with the published dataset. 

@fmassa what are your thoughts?

Reply:
In the one hand, I agree with @alykhantejani on both points.
In the other hand, the current state of things forces the user to manually replace/subtract 1 from the labels when they are training their models (as the losses supposes that the labels start from 0).
If we want the interface of this dataset to be a direct replacement for `mnist`, I'd go with this patch. But in any case, this should be documented somewhere.

Reply:
@fmassa That's a good point, I guess if we leave it as is (0 assigned label 10) then training wont work out of the box, which I think it important. Should we add a warning that the label has been changed, or just in the documentation?